src/read.c

```diff
@@ -3958,11 +3923,56 @@ static avifResult avifDecoderDecodeTiles(avifDecoder * decoder,
             }
         }
 
-        ++*decodedTileCount;
+        ++tileGroup->decodedTileCount;
     }
     return AVIF_RESULT_OK;
 }
 
+avifResult avifDecoderOutputDecodedTiles(avifDecoder * decoder, avifTileGroup * group, avifBool alpha, unsigned int oldDecodedTileCount)
+{
+    if (group->decodedTileCount == oldDecodedTileCount) {
+        // Nothing changed.
+        return AVIF_RESULT_OK;
+    }
+    // There is at least one newly decoded tile.
+    assert(group->decodedTileCount > oldDecodedTileCount);
+
+    // Grid path.
+    if ((group->grid.rows > 0) && (group->grid.columns > 0)) {
+        assert(group->tileCount == (group->grid.rows * group->grid.columns));
+        AVIF_CHECKERR(avifDecoderDataFillImageGrid(decoder->data, decoder->image, group, oldDecodedTileCount, alpha),
+                      AVIF_RESULT_INVALID_IMAGE_GRID);
+        return AVIF_RESULT_OK;
+    }
+
+    // Normal (most common) non-grid path.
+    assert(group->tileCount == 1);
+    avifImage * src = decoder->data->tiles.tile[group->firstTileIndex].image;
+    const avifPlanesFlag planes = alpha ? AVIF_PLANES_A : AVIF_PLANES_YUV;
+    if ((decoder->image->width != src->width) || (decoder->image->height != src->height) || (decoder->image->depth != src->depth)) {
+        if (alpha) {
+            avifDiagnosticsPrintf(&decoder->diag, "decoder->image does not match srcAlpha in width, height, or bit depth");
+            return AVIF_RESULT_DECODE_ALPHA_FAILED;
+        }
+        avifImageFreePlanes(decoder->image, AVIF_PLANES_ALL);
+
+        decoder->image->width = src->width;
+        decoder->image->height = src->height;
+        decoder->image->depth = src->depth;
+    }
+#if 0
+    // This code is currently unnecessary as the CICP is always set by the end of avifDecoderParse().
```

Comment:
is that going to change or instead of the code should just a note be left here?

Reply:
This code was only moved in this PR, not changed.